File: advancedscreenswitchersample/src/main/java/com/art/advancedscreenswitchersample/SampleTransitionAnimationProvider.java
package com.art.advancedscreenswitchersample;

import com.art.alligator.AnimationData;
import com.art.alligator.TransitionAnimationProvider;
import com.art.alligator.Screen;
import com.art.alligator.animations.TransitionAnimation;
import com.art.alligator.TransitionType;

/**
 * Date: 28.02.2017
 * Time: 21:01
 *
 * @author Artur Artikov
 */

public class SampleTransitionAnimationProvider implements TransitionAnimationProvider {
	@Override
	public TransitionAnimation getAnimation(TransitionType transitionType, Class<? extends Screen> screenClassFrom, Class<? extends Screen> screenClassTo, boolean isActivity, AnimationData animationData) {
		if(isActivity) {
			return TransitionAnimation.DEFAULT;
		} else {
			return new TransitionAnimation(R.anim.stay, R.anim.fade_out);
		}
	}
}

File: singleactivityflowsample/src/main/java/exp/terenfear/singleactivitynavigation/SampleTransitionAnimationProvider.kt
package exp.terenfear.singleactivitynavigation

import me.aartikov.alligator.DestinationType
import me.aartikov.alligator.Screen
import me.aartikov.alligator.TransitionType
import me.aartikov.alligator.animations.AnimationData
import me.aartikov.alligator.animations.SimpleTransitionAnimation
import me.aartikov.alligator.animations.TransitionAnimation
import me.aartikov.alligator.animations.providers.TransitionAnimationProvider


class SampleTransitionAnimationProvider : TransitionAnimationProvider {
    override fun getAnimation(transitionType: TransitionType, destinationType: DestinationType,
                              screenClassFrom: Class<out Screen>, screenClassTo: Class<out Screen>,
                              animationData: AnimationData?): TransitionAnimation {
        return if (destinationType == DestinationType.ACTIVITY) {
            TransitionAnimation.DEFAULT
        } else {
            SimpleTransitionAnimation(android.R.anim.fade_in, android.R.anim.fade_out)
        }
    }
}